### ISSUE TITLE
Add Blacklist for deprecations

### ISSUE THREAD

**Issue by slde-robin** [sensiolabs-de/deprecation-detector#7]
If your project uses a deprecation heavily your whole output is full of it.
We should add a configuration to blacklist some deprecations.

**Comment by umulmrum** [comment]
It might also be useful to be able to exclude certain files or directories.

**Comment by chr-hertel** [comment]
i agree on excluding certain files or directories, but i'm not sure about a deprecation blacklist.
I don't like the idea of extensive option usage like #17 provides as solution. maybe a blacklist by config file?
a config file would make it easy to exclude files and directories as well.

**Comment by bestform** [comment]
The implementation in #17 provides an option to implement a file/folder exclude as well. How you define the filter options can be decided for every filter separately. I think, the option to exclude certain calls is suited for being configured on the command line, but I am open for discussion on that matter.
The reason I think this is a good option is that the normal use case will probably be:

* run the checker
* see excessive use of one specific, deprecated method call
* filter that call out on a consecutive run to get a better overview of the remaining problems

Adding such filter to a config file wouldn't be a convenient way to go, and moreover would probably encourage ignoring certain errors by default and forgetting about it.

**Comment by MarvinKlemp** [comment]
after #17 we still need the implementation for other violations. May we should also add an output if a specific deprecation is filtered.

**Comment by bestform** [comment]
Adding active filters to the output is a good idea.

**Comment by MarvinKlemp** [comment]
@bestform do you want to add the missing filters ?

**Comment by bestform** [comment]
@MarvinKlemp what do you mean exactly? The filters itself (if so: is there a list of missing filters?) or the information about active filters in the output?

**Comment by MarvinKlemp** [comment]
if you want both :laughing: 
But IMO we should wait until #45 is merged
